tests/foreman/cli/test_fact.py

```diff
@@ -61,3 +61,56 @@ def test_negative_list_by_name(module_target_sat):
     assert (
         module_target_sat.cli.Fact().list(options={'search': f'fact={gen_string("alpha")}'}) == []
     )
+
+
+def test_positive_custom_facts(module_target_sat):
+    """Create custom facts and verify they are updated on Satellite
+
+    :id: ae0b5574-cc8b-4f0c-ba7b-c6f480c08e06
+
+    :steps:
+        1. Create few custom facts
+        2. Verify on Satellite
+
+    :expectedresults: Custom facts are created and uploaded successfully.
+    """
+    ptable = module_target_sat.cli_factory.make_partition_table()
+    os = module_target_sat.cli_factory.make_os()
+    module_target_sat.cli.PartitionTable.add_operating_system(
+        {'name': ptable['name'], 'operatingsystem': os['title']}
+    )
+    ptable = module_target_sat.cli.PartitionTable.info({'name': ptable['name']})
+    assert os['title'] in ptable['operating-systems']
```

Comment:
How does this function test the `Facts` component? What are the `few custom facts` in the context of the test?

Reply:
I have updated the test.